zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryRequestMapper.java

```diff
@@ -120,29 +131,185 @@ public static Either<ProblemDetail, ProcessInstanceSort> toSearchQuerySort(
         } else if ("endDate".equals(field)) {
           builder.endDate();
         } else {
-          throw new RuntimeException("unkown sortBy " + field);
+          // Collect validation error for unknown sortBy field
+          validationErrors.add(
+              ProblemDetail.forStatusAndDetail(
+                  HttpStatus.BAD_REQUEST, "Unknown sortBy field: " + field));
+          continue;
         }
 
         if ("asc".equalsIgnoreCase(order)) {
           builder.asc();
         } else if ("desc".equalsIgnoreCase(order)) {
           builder.desc();
         } else {
-          throw new RuntimeException("unkown sortOrder " + order);
+          // Collect validation error for unknown sortOrder
+          validationErrors.add(
+              ProblemDetail.forStatusAndDetail(
+                  HttpStatus.BAD_REQUEST, "Unknown sortOrder: " + order));
         }
       }
 
+      if (!validationErrors.isEmpty()) {
+        return Either.left(
+            validationErrors.get(0)); // Return the first encountered error for simplicity
```

Comment:
🔧 We can collect the validation errors as a `List<String>` instead and join them in one `ProblemDetail` in the end. Have a look at `RequestMapper::validateUpdateRequest` for reference. That way, all validation errors can be shown to the user at once. Otherwise, collecting them is useless.

Reply:
We should even go one step further and return an `Either<List<String>, XYZ>` to collect validation errors up the chain. Only the outermost caller should create the `ProblemDetail` in the end, e.g. `toUserTaskQuery`.

Reply:
I have update this on the follow commits: 
- https://github.com/camunda/camunda/pull/19625/commits/a8c8a4b72fa4854b0cc70d1bec06dcddc4da2f05 
- https://github.com/camunda/camunda/pull/19625/commits/55bd74cf1ca948c641ad37081c3cf2675c85209f